tests/HooksDeleteUserTest.php

```diff
@@ -47,79 +70,115 @@ protected function setUp(): void {
-				'app',
-				$activity['subject'],
-				json_encode([]),
-				'',
-				json_encode([]),
-				'file',
-				'link',
-				'user',
-				$activity['affectedUser'],
-				time(),
-				IExtension::PRIORITY_MEDIUM,
-				'test',
-			]);
-			$queryMailQueue->execute([
-				'app',
-				$activity['subject'],
-				json_encode([]),
-				$activity['affectedUser'],
-				time(),
-				'test',
-				time() + 10,
-			]);
+			$this->insertActivity($activity);
+			$this->insertActivityMQ($activity);
 		}
 	}
 
+	/**
+	 * @throws Exception
+	 */
 	protected function tearDown(): void {
-		$data = new Data(
-			$this->createMock(IManager::class),
-			\OC::$server->getDatabaseConnection()
-		);
-		$data->deleteActivities([
+		$this->data->deleteActivities([
 			'type' => 'test',
 		]);
-		$query = \OC::$server->getDatabaseConnection()->prepare("DELETE FROM `*PREFIX*activity_mq` WHERE `amq_type` = 'test'");
-		$query->execute();
+		$qb = $this->dbConnection->getQueryBuilder();
+		$qb->delete('activity_mq')
+			->where($qb->expr()->eq('amq_type', $qb->createNamedParameter('test')));
+		$qb->executeStatement();
 
 		parent::tearDown();
 	}
 
+	/**
+	 * @throws Exception
+	 */
 	public function testHooksDeleteUser(): void {
 		$this->assertUserActivities(['delete', 'otherUser']);
 		$this->assertUserMailQueue(['delete', 'otherUser']);
-		Hooks::deleteUser(['uid' => 'delete']);
+		$user = $this->createMock(IUser::class);
+		$user->expects($this->exactly(2))->method('getUID')->willReturn('delete');
+		$userDeleted = new UserDeleted($this->data, $this->mailQueueHandler);
+		$userDeleted->handle(new UserDeletedEvent($user));
 		$this->assertUserActivities(['otherUser']);
 		$this->assertUserMailQueue(['otherUser']);
 	}
 
+	/**
+	 * @throws Exception
+	 */
 	protected function assertUserActivities(array $expected): void {
-		$query = \OC::$server->getDatabaseConnection()->prepare("SELECT `affecteduser` FROM `*PREFIX*activity` WHERE `type` = 'test'");
+		$qb = $this->dbConnection->getQueryBuilder();
+		$query = $qb->select('affecteduser')
+			->from('activity')
+			->where($qb->expr()->eq('type', $qb->createNamedParameter('test')));
 		$this->assertTableKeys($expected, $query, 'affecteduser');
 	}
 
+	/**
+	 * @throws Exception
+	 */
 	protected function assertUserMailQueue(array $expected): void {
-		$query = \OC::$server->getDatabaseConnection()->prepare("SELECT `amq_affecteduser` FROM `*PREFIX*activity_mq` WHERE `amq_type` = 'test'");
+		$qb = $this->dbConnection->getQueryBuilder();
+		$query = $qb->select('amq_affecteduser')
+			->from('activity_mq')
+			->where($qb->expr()->eq('amq_type', $qb->createNamedParameter('test')));
 		$this->assertTableKeys($expected, $query, 'amq_affecteduser');
 	}
 
-	protected function assertTableKeys(array $expected, IPreparedStatement $query, string $keyName): void {
-		$query->execute();
+	/**
+	 * @throws Exception
+	 */
+	protected function assertTableKeys(array $expected, IQueryBuilder $query, string $keyName): void {
```

Comment:
Wanted to remove this test file as well but not sure what's being asserted in this function.

Reply:
Whether entries are actually deleted. I would say your other tests cover this sufficient.

Reply:
I'll remove the file then.